packages/session-replay-browser/src/session-replay.ts

```diff
@@ -166,25 +189,28 @@ export class SessionReplay implements AmplitudeSessionReplay {
   }
 
   getShouldRecord(ignoreFocus = false) {
-    if (!this.identifiers || !this.config) {
+    if (!this.identifiers || !this.config || !this.identifiers.sessionId) {
       this.loggerProvider.error(`Session is not being recorded due to lack of config, please call sessionReplay.init.`);
       return false;
     }
+
+    if (!this.config.captureEnabled) {
```

Comment:
[nit] kinda wish "config" based decisions can be grouped together (i.e. captureEnabled + sampling being evaluated together) and separate from other things.